tests/expressions_case/tests.py

```diff
@@ -312,6 +312,30 @@ def test_in_subquery(self):
             transform=attrgetter('integer', 'integer2')
         )
 
+    def test_exists(self):
+        instance = CaseTestModel.objects.annotate(
+            annotation=Case(
+                When(Exists(FKCaseTestModel.objects.filter(fk=OuterRef('pk'), integer=2)), then=Value("2")),
+                When(Exists(FKCaseTestModel.objects.filter(fk=OuterRef('pk'), integer=1)), then=Value("1")),
+                default=Value(""),
+                output_field=CharField(),
+            ),
+        ).get(integer=1)
+        self.assertEquals(instance.annotation, "1")
+
+    def test_exists_with_condition(self):
+        instance = CaseTestModel.objects.annotate(
+            annotation=Case(
+                When(Exists(FKCaseTestModel.objects.filter(fk=OuterRef('pk'), integer=1)), integer2=2,
+                     then=Value("2")),
+                When(Exists(FKCaseTestModel.objects.filter(fk=OuterRef('pk'), integer=1)), integer2=1,
+                     then=Value("1")),
+                default=Value(""),
+                output_field=CharField(),
+            ),
+        ).get(integer=1)
+        self.assertEquals(instance.annotation, "1")
```

Comment:
`assertEquals` is deprecated in Python 3. Also not sure if this test belongs in `expressions.tests`.

Reply:
Ok.

You're right, in light of the Exists test being in `expressions.tests`, this should probably go there too in its current form.

But I'm also not really trying to test `Exists` here, that just my use case. The real test should be using a query argument with a keyword argument, so I will modify to be simpler.

Reply:
You could pass a literal boolean using `Value(True, output_field=models.BooleanField())`.